tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/IncClassEmitter.scala

```diff
@@ -0,0 +1,77 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___      __ ____  Scala.js tools             **
+**    / __/ __// _ | / /  / _ | __ / // __/  (c) 2016, LAMP/EPFL        **
+**  __\ \/ /__/ __ |/ /__/ __ |/_// /_\ \    http://scala-js.org/       **
+** /____/\___/_/ |_/____/_/ | |__/ /____/                               **
+**                          |/____/                                     **
+\*                                                                      */
+
+package org.scalajs.core.tools.linker.backend.emitter
+
+import scala.collection.mutable
+
+import org.scalajs.core.tools.sem._
+
+import org.scalajs.core.tools.linker.backend.OutputMode
+import org.scalajs.core.tools.linker.LinkingUnit
+
+/** Tracks incrementality breaking changes made during class emitting.
+ *
+ *  The only reason this is not `private[emitter]` is because `RhinoJSEnv`
+ *  needs it.
+ */
+private[scalajs] final class IncClassEmitter(semantics: Semantics,
+    outputMode: OutputMode, internalOptions: InternalOptions) {
+  import IncClassEmitter._
+
+  def this(semantics: Semantics, outputMode: OutputMode) =
+    this(semantics, outputMode, InternalOptions())
```

Comment:
Rhino i.e. ScalaJSCoreLib.scala there was a second constructor in ScalaJSClassEmitter and I removed it and "moved" it here as Rhino is no longer directly accessing to ScalaJSClassEmitter's constructor but creating a IncClassEmitter instead.